t/instant_answer_basics.t

```diff
@@ -0,0 +1,110 @@
+use strict;
+use warnings;
+
+# Database setup
+BEGIN {
+    $ENV{DDGC_DB_DSN} = 'dbi:SQLite:dbname=ddgc_test.db';
+}
+
+use Test::More;
+use t::lib::DDGC::TestUtils;
+use HTTP::Request::Common;
+use Plack::Test;
+use Plack::Builder;
+use Plack::Session::State::Cookie;
+use Plack::Session::Store::File;
+use File::Temp qw/ tempdir /;
+use JSON::MaybeXS qw/:all/;
+
+use DDGC;
+use DDGC::Web;
+
+my $d = DDGC->new;
+t::lib::DDGC::TestUtils::deploy( undef, $d->db );
+
+# Application setup and creation
+my $app = builder {
+    enable 'Session',
+        store => Plack::Session::Store::File->new(
+            dir => tempdir,
+        ),
+        state => Plack::Session::State::Cookie->new(
+            secure => 0,
+            httponly => 1,
+            expires => 21600,
+            session_key => 'ddgc_session',
+        );
+    mount '/testutils' => t::lib::DDGC::TestUtils->to_app;
+    mount '/' => DDGC::Web->new->psgi_app;
+};
+
+# Tests - assert things here
+test_psgi $app => sub {
+    my ( $cb ) = @_;
+
+    # Create user
+    my $user_request = $cb->(
+        POST '/testutils/new_user',
+        { username => 'ia_user' }
+    );
+    ok( $user_request->is_success, 'Creating an IA user' );
+
+    # Get session
+    my $session_request = $cb->(
+        POST '/testutils/user_session',
+        { username => 'ia_user' }
+    );
+    ok( $session_request->is_success, 'Getting IA user Cookie' );
```

Comment:
We get a [HTTP::Response](https://metacpan.org/pod/HTTP::Response) here - chock full of useful stuff.